src/pages/index.js

```diff
@@ -6,78 +7,224 @@ import PopupWithImage from "../components/PopupWithImage.js";
+    .then((data) => {
+      const cardElement = createCard(data);
+      cardSection.addItem(cardElement);
+      addCardForm.close();
+      addCardForm.reset();
+      addCardValidator._disableButton();
+      addCardForm.close();
+    })
+    .catch((error) => {
+      console.error("Error adding card:", error);
+    })
+    .finally(() => {
+      addCardForm.renderLoading(false);
+    });
+}
+
+const addCardForm = new PopupWithForm(
+  "#add-card-modal",
+  handleAddCardFormSubmit
+);
+addCardForm.setEventListeners();
+
+// EVENT LISTENER FOR NEW CARD BUTTON
+constants.addCardButton.addEventListener("click", () => {
+  addCardForm.open();
+  addCardValidator._disableButton();
+});
+
+// PREVIEW IMAGE MODAL
 const cardPreview = new PopupWithImage("#preview-image-modal");
 cardPreview.setEventListeners();
 
 function handleImageClick(name, link) {
   cardPreview.open({ name, link });
 }
 
-function createCard(cardData) {
-  const cardElement = new Card(cardData, "#card-template", handleImageClick);
-  return cardElement.getView();
+const deleteCardPopup = new PopupWithForm("#delete-card-modal", () => {
+  api
+    .deleteRequest(currentCardID)
+    .then(() => {
+      currentCardElement.remove();
+      deleteCardPopup.close();
+    })
+    .catch((error) => console.error("Error deleting card:", error));
+});
+
+let currentCardID = null;
+let currentCardElement = null;
+
+function handleDeleteCard(cardID, cardElement) {
+  currentCardID = cardID;
+  currentCardElement = cardElement;
+  deleteCardPopup.open();
 }
 
-// Initialize user information
-const userInformation = new UserInfo({
-  name: ".profile__title",
-  description: ".profile__description",
-});
+//-----HANDLE LIKES----/
+
+function handleLikeButton(likeButton, likedStatus, cardID) {
+  if (likedStatus) {
+    return api
+      .removeLike(cardID)
+      .then(() => {
+        likeButton.classList.remove("card__like-button_active");
+        return false;
+      })
+      .catch((error) => console.error("Error removing like:", error));
+  } else {
+    return api
+      .addLike(cardID)
+      .then(() => {
+        likeButton.classList.add("card__like-button_active");
+        return true;
+      })
+      .catch((error) => console.error("Error adding like:", error));
+  }
+}
+//----------USER INFO------//
+
+const userInfo = new UserInfo(
+  ".profile__title",
+  ".profile__description",
+  ".profile__image"
+);
+
+api
+  .getUser()
+  .then((inputValues) => {
+    userInfo.setUserInfo(inputValues.name, inputValues.about);
+    userInfo.setAvatar(inputValues.avatar);
+  })
+  .catch((err) => console.error("Error fetching user data:", err));
+
+function handleProfileEditSubmit(inputValues) {
+  profileEditForm.renderLoading(true);
+  api
+    .editProfile(inputValues.name, inputValues.about)
+    .then((data) => {
+      userInfo.setUserInfo(data.name, data.about);
+      profileEditForm.close();
+    })
+    .catch((err) => console.error("Error editing profile:", err))
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors